openstf/model/train.py

```diff
@@ -160,30 +243,93 @@ def train_model_for_specific_pj(
         model_trainer.confidence_interval,
     )
     train_predict = predictor.make_forecast(
-        train_data.iloc[:, 1:].drop("Horizon", axis=1, errors="ignore")
+        split_input_data["train_data"]
+        .iloc[:, 1:]
+        .drop("Horizon", axis=1, errors="ignore")
     )
     validation_predict = predictor.make_forecast(
-        validation_data.iloc[:, 1:].drop("Horizon", axis=1, errors="ignore")
+        split_input_data["validation_data"]
+        .iloc[:, 1:]
+        .drop("Horizon", axis=1, errors="ignore")
     )
     test_predict = predictor.make_forecast(
-        test_data.iloc[:, 1:].drop("Horizon", axis=1, errors="ignore")
+        split_input_data["test_data"]
+        .iloc[:, 1:]
+        .drop("Horizon", axis=1, errors="ignore")
     )
 
-    context.perf_meter.checkpoint("predicting")
+    return {
+        "train_predict": train_predict,
+        "validation_predict": validation_predict,
+        "test_predict": test_predict,
+    }
+
+
+def create_evaluation_figures_for_specific_pj(
+    model_trainer, split_input_data, split_predicted_data
+):
+    """Create figures of feature importance and data series of input and predicted data.
+
+    Args:
+        model_trainer (MLModelType): model trainer
+        split_input_data (dict): dict of cleaned data with features, split in train,
+            validation and test
+        split_predicted_data (dict): dict of predicted data, split in train, validation
+            and test
+
+    Returns:
+        tuple: tuple containing:
+            plotly.graph_objects.Figure: A treemap of the features.
+            dict: dict of line plots of of split input and predicted
+                data series per prediction horizon.
+    """
 
     # Create figures
     figure_features = plot_feature_importance(model_trainer.feature_importance)
     figure_series = {
```

Comment:
Perhaps this is all a bit too much for readability. You're doing dictionary comprehension, key interpolation, calling a function with nested calls as arguments while looping over a list which is constructed though some chained function call as well. I would suggest if you want to use the dict comprehension that you construct the variables first:

```python
    horizons = split_input_data["train_data"].Horizon.unique()
    input_data_list = list(split_input_data.values())
    predicted_data_list = list(split_predicted_data.values())

    figure_series = {
        f"Predictor{horizon}": plot_data_series(
            data=input_data_list, predicted_data=predicted_data_list, horizon=horizon
        ) for horizon in horizons
    }
```

Reply:
I did not look into this too much, but agree the complexity in one line is increased. I followed your suggestion.